### ISSUE TITLE
Serve documents with the correct mime type when using a non-local storage backend

### ISSUE THREAD

**Issue by nilssolanki** [wagtail/wagtail#1416]
In my models, I have a `wagtaildocs.Document` field:

```
svg_mask = models.ForeignKey(
    verbose_name=u"Mask (SVG)",
    to='wagtaildocs.Document',
    null=True,
    blank=True,
    on_delete=models.SET_NULL,
    related_name='+'
)
```

I would like to serve an svg as an `image/svg+xml`, not as a stream, when accessing it via its URL. So far, I got the following advice (https://github.com/torchbox/wagtail/issues/1415#issuecomment-112769092):

> The streaming behaviour isn't dictated by the mime type.
> You can switch it off globally though by installing `django-sendfile` and adding the following to your settings:
> `SENDFILE_BACKEND = 'sendfile.backends.simple'`

Unfortunately, this did not seem to prevent the streaming. Do you know how I could save this?

**Comment by gasman** [comment]
Several layers of confusion here, sorry!

"Serving as a stream" means something different to us - for us, it means the response is generated in stages, rather than being constructed in memory up front. You're talking about the mime type `application/octet-stream`, which is unrelated to that behaviour.

We made several changes to document serving in 1.0b2, including serving them with the correct mime type - however, we've now discovered that these changes are incompatible with remote storage backends such as `storages.backends.s3boto.S3BotoStorage`. This causes the error you saw in #1415. For now, I've fixed this (in #1417) so that these backends receive the old (1.0b1) behaviour, including the `application/octet-stream` mime type. We'll try to come up with an improved fix in Wagtail 1.1, so that remote backends use the correct mime type too.

**Comment by nilssolanki** [comment]
Oh I see :) That makes sense, thank you.

**Comment by gasman** [comment]
Will be addressed by #1420

**Comment by gasman** [comment]
Completed in #5296 (which changed the default behaviour of remote storage backends to redirect to the remote URL rather than proxying through Django, meaning that headers are whatever the storage service sends).
